File: code/dbrunken/Python/make_change.py
'''
create monetary change
'''
from typing import Counter


coins = [
    ("half dollar", .5),
    ('quarter', .25),
    ('dime', .10),
    ('nickle', .05),
    ('penny',  .01)
]

'''
pass in total dollar amount from input
convert to pennies
record running sum
divide total by 25(quarters)
'''

def make_change(user_in):
    pennies = user_in * 100
    # print(round(pennies, 2))
    count = 0
    for coin in coins:
        coinConvert = coin[1] * 100 # convert to pennies
        print(pennies)
        coinCount =  int(pennies) // int(coinConvert) # floor divide to determine how many pennies there are
        pennies -= coinCount * coinConvert # subtract remaining pennies from total
        # print(count)
        print(coinCount, coins[count][0])
        count +=1
    total = print(f'\n is what you have when {user_in} is converted to pennies. {coinCount} pennies is the remainder you will have with {coins[0][0]}.')

    return total
    




total = input('how much money would you like to make change out of? ')
total = float(total)

make_change(total)


# if total == 0:
#     print("i see no money")

# else:
   

File: code/dbrunken/JavaScript/Labs/lab1/make change/part 1/make_change.js
/*
make change
*/

const coins = {
    quarter: 25,
    dime: 10,
    nickel: 5,
    penny: 1
}

const coinNames = ['quarter', 'dime', 'nickel', 'penny']
for (let i = 0; i< coinNames.length; i ++) {
    let coin = coinNames[i]
    let coinValue = coins[coin]
    console.log(coin, coinValue)
}

let usersChange = prompt("enter change amount")


function change(userChange) {
    let pennies = userChange * 100
    Math.floor(pennies)

    let quarters = Math.floor(pennies / 25)
    pennies = pennies - (quarters*25)
    
    let dimes = Math.floor(pennies/10)
    pennies = pennies - (dimes*10)

    let nickels = Math.floor(pennies/5)
    pennies = pennies - (nickels*5)

    return {quarters, dimes, nickels, pennies}
}
const changeData = change(usersChange)
alert(`${changeData.quarters}` + " quarters remain.")
alert(`${changeData.dimes}` + " dimes remain.")
alert(`${changeData.nickels}` + " nickles remain.")
alert(`${changeData.pennies}` + " pennies remain.")



/*
step 1: user enters dollar change amount
step 2: convert amount by 100 to get all pennies
step 3: floor divide to isolate remainder
step 4: with remainder, divide by highest coin
step 5: continue to ==0
return total and 'alert'
*/